presto-tests/src/test/java/io/prestosql/tests/AbstractTestEngineOnlyQueries.java

```diff
@@ -111,6 +113,25 @@ public void testTimeLiterals()
         // TODO assertQuery(kathmandu, "SELECT TIME '3:04:05'");
     }
 
+    /**
+     * Tests query runner with results of vaious precisions, and query assert.
+     */
+    @Test
+    public void testTimeQueryResult()
+    {
+        assertThat(query("SELECT TIME '01:23:45.123'")).matches("SELECT TIME '01:23:45.123'");
+        assertThat(query("SELECT TIME '01:23:45.123456'")).matches("SELECT TIME '01:23:45.123456'");
+        assertThat(query("SELECT TIME '01:23:45.123456789'")).matches("SELECT TIME '01:23:45.123456789'");
+        assertThat(query("SELECT TIME '01:23:45.123456789012'")).matches("SELECT TIME '01:23:45.123456789012'");
+
+        QueryAssert queryAssert = assertThat(query("SELECT TIME '01:23:45.123456789012'"));
```

Comment:
Why is this outside the lambda below and as a separate variable?

Reply:
So that `assertThatThrowBy` closure is as small as possible.